generators/frontend/templates/react-demo/components/CategoriesList.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import CategoryCard from './CategoryCard';
+import Util from '../common/Util';
+import styled from 'styled-components';
+
+export default class extends Component {
+  state = {
+    categories: [],
+    productsList: [],
+    show: true
+  };
+
+  async componentDidMount() {
+    const cats = await Util.fetchCategories();
+    const prods = await Util.fetchProducts(cats[0].id);
+    this.setState(
+      {
+        categories: cats,
+        productsList: prods.results
+      }
+    );
+  }
+
+  onClickCategory = (e, categoryId) => {
+    e.preventDefault();
+    Util.fetchProducts(categoryId).then(list =>
+      this.setState({ productsList: list.results })
+    );
+  }
+
+  showProductsList = (e) => {
+    e.preventDefault();
+    this.setState({ show: false })
+  }
+
+  render() {
+    const { categories, productsList } = this.state;
+
+    const Wrapper = styled.div`
```

Comment:
Styled components should be defined outside `render` method. Check this code snippet as an example: https://gist.github.com/anonymous/be700e21646e08f3b410fe6e0195ef89#file-themeselect-jsx